src/commands/Misc/reddituser.ts

```diff
@@ -0,0 +1,219 @@
+			.addField(titles.BEST_COMMENT,
+				[
+					`${bestComment.subreddit} - **${bestComment.score}**`,
+					`${message.language.duration((Date.now() - bestComment.created) * 1000)} ago`,
+					`[${fieldsData.PERMALINK}](https://reddit.com${bestComment.permalink})`,
+					cutText(bestComment.body, 900)
+				].join('\n'))
+			.addField(titles.WORST_COMMENT,
+				[
+					`${worstComment.subreddit} - **${worstComment.score}**`,
+					`${message.language.duration((Date.now() - worstComment.created) * 1000)} ago`,
+					`[${fieldsData.PERMALINK}](https://reddit.com${worstComment.permalink})`,
+					cutText(worstComment.body, 900)
+				].join('\n'));
+
+		return message.sendEmbed(embed);
+	}
+
+	private async fetchData(user: string, message: KlasaMessage) {
+		return Promise.all([this.fetchAbout(user, message), this.fetchComments(user, message), this.fetchSubmissions(user, message)]);
+	}
+
+	private async fetchAbout(user: string, message: KlasaMessage) {
+		const { data } = await fetch(`https://www.reddit.com/user/${user}/about/.json`, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'about'>;
+		return data;
+	}
+
+	private async fetchComments(user: string, message: KlasaMessage, after = '', dataElements: Reddit.CommentDataElement[] = []) {
+		const url = new URL(`https://www.reddit.com/user/${user}/comments.json`);
+		url.searchParams.append('after', after);
+		url.searchParams.append('limit', '100');
+
+		const { data } = await fetch(url, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'comments'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchComments(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private async fetchSubmissions(user: string, message: KlasaMessage, after = '', dataElements: Reddit.PostDataElement[] = []) {
+		const url = new URL(`https://www.reddit.com/user/${user}/submitted.json`);
+		url.searchParams.append('after', after);
+		url.searchParams.append('limit', '100');
+
+		const { data } = await fetch(url, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'posts'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchSubmissions(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private calculateControversiality(comments: Reddit.CommentDataElement[]) {
+		if (!comments.length || comments.length < 5) return 0;
+		let count = 0;
+
+		for (const comment of comments) {
+			if (comment.controversiality === 1) count++;
+		}
+
+		return (count / comments.length) * 100;
+	}
```

Comment:
This method needs to be documented clarifying what exactly the algorithm does and what each condition means.

Reply:
I added the comment `Calculated the controversiality based on whether the Reddit has marked a comment controversial` but it should be noted that controversiality is a reddit thing and not necessarily algorithmic. This is just a basic percentile calculation and the API can have some comments marked as controversial. Reddit defines controversial as: **Controversial are posts or comments that have very even amounts of up/downvotes and a lot of activity.**

Reply:
Update: added this:

```
/**
 * Calculated the controversiality based on whether the Reddit has marked a comment controversial
 * Controversial posts or comments have very even amounts of up/downvotes and a lot of activity.
 * @param comments The reddit comments to take into the calculation
 */
```